aea/aea.py

```diff
@@ -338,6 +338,9 @@ def handle_envelope(self, envelope: Envelope) -> None:
         if msg is None:
             return
 
+        if envelope.context is not None:
```

Comment:
I got the idea but don't like the way of the extending message class.

Reply:
Yeah there is a downside to it. However, any alternative approach ends up becoming quite verbose...

Reply:
removed